var/spack/repos/builtin/packages/moab/package.py

```diff
@@ -55,61 +79,119 @@ class Moab(Package):
-            'CC=%s' % spec['mpi'].mpicc,
-            'FC=%s' % spec['mpi'].mpifc]
+            '--without-vtk'
+        ]
+        if '+mpi' in spec:
+            options.extend([
+                '--with-mpi=%s' % spec['mpi'].prefix,
+                'CXX=%s' % spec['mpi'].mpicxx,
+                'CC=%s' % spec['mpi'].mpicc,
+                'FC=%s' % spec['mpi'].mpifc
+            ])
+            if '+parmetis' in spec:
+                options.append('--with-parmetis=%s' % spec['parmetis'].prefix)
+            else:
+                options.append('--without-parmetis')
+#          FIXME: --without-mpi does not configure right
+#       else:
+#           options.append('--without-mpi')
+
+        if '+hdf5' in spec:
+            options.append('--with-hdf5=%s' % spec['hdf5'].prefix)
+        else:
+            options.append('--without-hdf5')
 
-        if '~fortran' in spec:
-            options.append('--disable-fortran')
-        if '+shared' in spec:
-            options.append('--enable-shared')
         if '+netcdf' in spec:
             options.append('--with-netcdf=%s' % spec['netcdf'].prefix)
+        else:
+            options.append('--without-netcdf')
+
+        if '+pnetcdf' in spec:
+            options.append('--with-pnetcdf=%s'
+                           % spec['parallel-netcdf'].prefix)
+        else:
+            options.append('--without-pnetcdf')
+
+        if '+cgm' in spec:
+            options.append('--with-cgm=%s' % spec['cgm'].prefix)
+            if '+irel' in spec:
+                options.append('--enable-irel')
+            else:
+                options.append('--disable-irel')
+        else:
+            options.append('--without-cgm')
+        if '+fbigeom' in spec:
+            options.append('--enable-fbigeom')
+        else:
+            options.append('--disable-fbigeom')
 
-        configure(*options)
-        make()
-        make('install')
+        if '+coupler' in spec:
+            options.append('--enable-mbcoupler')
+        else:
+            options.append('--disable-mbcoupler')
+
+        if '+metis' in spec:
+            options.append('--with-metis=%s' % spec['metis'].prefix)
+        else:
+            options.append('--without-metis')
+
+        if '+parmetis' in spec:
+            options.append('--with-parmetis=%s' % spec['parmetis'].prefix)
+        else:
+            options.append('--without-parmetis')
+
+        if '+zoltan' in spec:
+            options.append('--with-zoltan=%s' % spec['zoltan'].prefix)
+        else:
+            options.append('--without-zoltan')
+
+        if '+debug' in spec:
+            options.append('--enable-debug')
+        else:
+            options.append('--disable-debug')
+
+        # FIXME it seems that with cgm and shared, we have a link
+        #   issue  in tools/geometry
+        if '+shared' in spec:
+            options.append('--enable-shared')
+        else:
+            options.append('--disable-shared')
+
+        if '~fortran' in spec:
```

Comment:
Might make sense to also check if indeed a fortran compiler is available as in `spack_fc` is non-empty or `self.compiler.fc` or some such thing.